src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -51,46 +51,42 @@ class Serializer implements SerializerInterface, NormalizerInterface, Denormaliz
      */
     public const EMPTY_ARRAY_AS_OBJECT = 'empty_array_as_object';
 
-    private const SCALAR_TYPES = [
-        'int' => true,
-        'bool' => true,
-        'float' => true,
-        'string' => true,
-    ];
+    private NormalizerInterface $normalizer;
+
+    private DenormalizerInterface $denormalizer;
 
     protected ChainEncoder $encoder;
 
     protected ChainDecoder $decoder;
 
-    /**
-     * @var array<string, array<string, array<bool>>>
-     */
-    private array $denormalizerCache = [];
-
-    /**
-     * @var array<string, array<string, array<bool>>>
-     */
-    private array $normalizerCache = [];
-
     /**
      * @param array<NormalizerInterface|DenormalizerInterface> $normalizers
      * @param array<EncoderInterface|DecoderInterface>         $encoders
      */
     public function __construct(
-        private array $normalizers = [],
+        array $normalizers = [],
         array $encoders = [],
+        ChainNormalizerInterface $normalizerAggregator = null,
```

Comment:
I understand the fact of adding optional arguments for the upgrade path, but what's the next step? How to get rid of the two first arguments?

Reply:
I would like all users to pass an empty array as first argument. Then at Symfony 8.0 we will just drop the first argument.

Reply:
Does the first argument drop will be possible from a BC point of view?

Reply:
No. That is why we need to do it in SF 8.0.